markdown.dtx

```diff
@@ -24774,6 +25060,64 @@ end
 %    \end{macrocode}
 % \begin{markdown}
 %
+%#### Line blocks
+%
+% The \luamdef{extensions.line_blocks} function implements the Pandoc line blocks
+% syntax extension.
+%
+% \end{markdown}
+%  \begin{macrocode}
+M.extensions.line_blocks = function()
+  return {
+    name = "built-in line_blocks syntax extension",
+    extend_writer = function(self)
+%    \end{macrocode}
+% \par
+% \begin{markdown}
+%
+% Define \luamdef{writer->lineblock} as a function that will transform
+% a line block consisted of `lines` to the output format, with
+% all but the last newline rendered as a line break.
+%
+% \end{markdown}
+%  \begin{macrocode}
+      function self.lineblock(lines)
+        if not self.is_writing then return "" end
+        local buffer = {}
+        for i = 1, #lines - 1 do
+          buffer[#buffer + 1] = { lines[i], self.linebreak }
+        end
+        buffer[#buffer + 1] = lines[#lines]
+        
+        return {"\\markdownRendererLineBlockBegin\n"
+                  ,buffer,
+                  "\n\\markdownRendererLineBlockEnd "}
+      end
+    end, extend_reader = function(self)
+      local parsers = self.parsers
+      local writer = self.writer
+
+      local LineBlock = Ct(
+                        (Cs(
+                          ( (parsers.pipe * parsers.space)/""
+                          * ((parsers.space)/entities.char_entity("nbsp"))^0
```

Comment:
There does not seem to be a good reason for hard-coding a non-breaking space here. We want TeX to obey spaces, which we can request via \obeyspaces inside a \begingroup ... \endgroup block. No need to touch spaces in Lua.

Reply:
Sorry, this has been a poor judgement on my part. We clearly only want to obey spaces at the beginning of a line, not in the middle of it. Using a non-breaking space does not seem clean from the semantics point of view, but it seems practical and I don't have a better suggestion, so let's keep it for now.

I replaced `entities.char_entity("nbsp")` for `writer.nbsp`, so that we don't have to rely on the writer rewriting the Unicode character to the correct TeX command (also, it is shorter). I also reverted the addition of `\obeyspaces` in 
acf0413. I am puzzled as to why the unit tests were passing before acf0413, because the `nbsp` in the output should have caused an error in the CI.